### PR TITLE
Add user agent to prompty standalone execution

### PR DESCRIPTION
**Pull request by singankit**
# Description

Add capability to pass extra headers when using prompty

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-evals/promptflow/evals/evaluators/_relevance/_relevance.py [hunk 1]
```diff
@@ -44,7 +48,13 @@ def __init__(self, model_config: AzureOpenAIModelConfiguration):
         if model_config.api_version is None:
             model_config.api_version = "2024-02-15-preview"
 
-        prompty_model_config = {"configuration": model_config}
+        prompty_model_config = {
+            "configuration": model_config,
+        }
+
+        prompty_model_config.update({"parameters": {"extra-headers": {"x-ms-user-agent": USER_AGENT}}})\
```
Comment (lalala123123) [inline src/promptflow-evals/promptflow/evals/evaluators/_relevance/_relevance.py:55]: If prompty_model_config["parameters"]["extra-headers"] has a value, the original value will be overwritten

Comment (singankit) [reply src/promptflow-evals/promptflow/evals/evaluators/_relevance/_relevance.py:55]: Agreed however the prompty used in this evaluator is not a user defined

**Review by nagkumar91 — APPROVED** [review]

File: src/promptflow-evals/promptflow/evals/evaluators/_relevance/_relevance.py (hunk 1 above)
Comment (singankit) [reply src/promptflow-evals/promptflow/evals/evaluators/_relevance/_relevance.py]: Prompty here is a part of relevance evaluator and it does not have this extra_headers param

**Review by lalala123123 — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Comment by singankit** [comment]
Fixed AB#3748666

**Pull request merged**